### PR TITLE
Run test in parallel

### PR DESCRIPTION
**Pull request by AJenbo**
Type: refactoring
Breaking change: no

Use brianium/paratest to run test in parallel across CPU threads. This greatly speeds up the time to run the full test suite from around 2 minutes to about 10 sec.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -27,7 +28,7 @@
         "psr-4": {"PDepend\\": "src/main/php/PDepend"}
     },
     "scripts": {
-        "test": "phpunit --no-coverage",
+        "test": "paratest --no-coverage",
```
Comment (ravage84) [inline composer.json:31]: Does that work under Windows, too?

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -46,7 +46,7 @@ jobs:
         composer update --prefer-dist --no-progress --prefer-${{ matrix.setup }}
 
     - name: Coverage
-      run: vendor/bin/phpunit --coverage-clover=coverage.xml --coverage-text --coverage-html=coverage
+      run: vendor/bin/paratest --coverage-clover=coverage.xml --coverage-text --coverage-html=coverage
```
Comment (ravage84) [inline .github/workflows/coverage.yml:49]: Can this be run in a Windows workflow VM, too?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -45,9 +45,13 @@ jobs:
...
 
       - name: Install dependencies
-        if: steps.composer-cache.outputs.cache-hit != 'true'
+        if: steps.composer-cache.outputs.cache-hit != 'true' && matrix.php != '8.4'
         run: composer update --prefer-dist --no-progress --prefer-${{ matrix.setup }}
 
+      - name: Install dependencies (PHP 8.4 beta)
+        if: steps.composer-cache.outputs.cache-hit != 'true' && matrix.php == '8.4'
+        run: composer update --prefer-dist --no-progress --prefer-${{ matrix.setup }} --ignore-platform-req=php
+
```
Comment (ravage84) [inline .github/workflows/tests.yml:54]: How is this related to the paratest changes?

Comment (AJenbo) [reply .github/workflows/tests.yml:54]: paratest is nor marked as PHP 8.4 compatible as 8.4 has not been released yet, so I needed to ignore the PHP version check using `--ignore-platform-req=php` for it to be installed. But I wanted to make it clear that this was an exception that applies just for 8.4 as long as it's in beta, rather then return to the old ways of just ignoring version checks in general.

Comment (ravage84) [reply .github/workflows/tests.yml:54]: Thanks for the explanation. Could we include a short explanation in the workflow file, may be?

At least, it's now explained in the related PR, here.

File: composer.json (hunk 1 above)
Comment (AJenbo) [reply composer.json:31]: Yep, ran as expected:
image

Went from 44 sec to 21 sec

File: .github/workflows/coverage.yml (hunk 2 above)
Comment (AJenbo) [reply .github/workflows/coverage.yml:49]: It probably could, but this workflow specifically prescribe Ubuntu: https://github.com/pdepend/pdepend/blob/3.x/.github/workflows/coverage.yml#L11

If you where to execute it on a Windows box it would fetch a Linux worker and run that in a VM. We can setup a Windows test in GitHub action if you like, but I think that should not be part of this PR.

Comment (ravage84) [reply .github/workflows/coverage.yml:49]: All good. Just wanted to make sure this change doesn't limit us in OS inclusiveness. 😆

File: .github/workflows/tests.yml (hunk 3 above)
Comment (AJenbo) [reply .github/workflows/tests.yml:55]: Added a comment on line 51, hope that works :)

Comment (ravage84) [reply .github/workflows/tests.yml:55]: Beautiful, thanks! 🙇‍♂️

**Review by ravage84 — APPROVED** [review]
